### ISSUE TITLE
immerMap function

### ISSUE THREAD

**Issue by achung89** [mweststrate/immer#35]
possibly add an immer implementation that can be used with map/filter/forEach
```
ex. 
const immerMap = (fn) => 
  (item) => immer( item, fn )

var arr = [['1'], ['2'], ['3']];
arr.map(immerMap((item) => {item.push('new')}));
```

**Comment by hex13** [comment]
this is why we could use of swap arguments and currying.

(proposal)
```javascript
arr.map(immer((item) => {item.push('new')}));
```

**Comment by mweststrate** [comment]
Implemented in 0.4.0. See https://github.com/mweststrate/immer#currying. The first example covers the above
